# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/WeakMap.cpp
# Commit: b1117df3b91d
# Full Hash: b1117df3b91d10d68d10b399637cf9ac295057e6
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2019-11-01 16:13:11
# Regressor Bug: 1587096
# File Overlap Count: 1
# Description:
#   Bug 1587096 - Part 3: Implmement FinalizationGroup r=sfink
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D49946
# ==============================================================================

diff -r cefc0474e13c -r b1117df3b91d js/src/gc/WeakMap.cpp
--- a/js/src/gc/WeakMap.cpp	Fri Nov 01 10:37:40 2019 +0000
+++ b/js/src/gc/WeakMap.cpp	Fri Nov 01 10:37:47 2019 +0000
@@ -182,6 +182,11 @@
   return true;
 }
 
+void ObjectWeakMap::remove(JSObject* key) {
+  MOZ_ASSERT(key);
+  map.remove(key);
+}
+
 void ObjectWeakMap::clear() { map.clear(); }
 
 void ObjectWeakMap::trace(JSTracer* trc) { map.trace(trc); }